### PR TITLE
[IR] Support Collection<*>.indices in for loop lowering.

### PR DESCRIPTION
**Pull request by madsager**
Additionally unmute a couple of tests that are working as intended.

### CODE REVIEW

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/ProgressionHandlers.kt [hunk 1]
```diff
@@ -247,7 +245,7 @@ internal abstract class IndicesHandler(protected val context: CommonBackendConte
 
 internal class ArrayIndicesHandler(context: CommonBackendContext) : IndicesHandler(context) {
```
Comment (punzki) [inline compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/ProgressionHandlers.kt]: Suggestion: Rename to `CollectionIndicesHandler`

File: compiler/testData/codegen/bytecodeText/forLoop/forInIndices/forInCollectionIndices.kt [hunk 2]
```diff
@@ -12,6 +11,16 @@ fun test() {
 // 0 getFirst
 // 0 getLast
 
+// JVM non-IR uses while.
+// JVM IR uses if + do-while.
+
+// JVM_TEMPLATES
 // 0 IF_ICMPGT
```
Comment (punzki) [inline compiler/testData/codegen/bytecodeText/forLoop/forInIndices/forInCollectionIndices.kt]: These two lines (`// 0 IF_...`) can be removed; they are made redundant by the `// 1 IF`

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/ProgressionHandlers.kt (hunk 1 above)
Comment (madsager) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/ProgressionHandlers.kt]: Good idea, done.

File: compiler/testData/codegen/bytecodeText/forLoop/forInIndices/forInCollectionIndices.kt (hunk 2 above)
Comment (madsager) [reply compiler/testData/codegen/bytecodeText/forLoop/forInIndices/forInCollectionIndices.kt]: Thanks. Done.

**Comment by udalov** [comment]
@madsager Thanks!

**Pull request merged**